translate/src/context/Editor.tsx

```diff
@@ -148,71 +115,111 @@ export type EditorActions = {
 
   setEditorSelection(content: string): void;
 
+  /** Set the result value of the active input */
+  setResultFromInput(idx: number, value: string): void;
+
   toggleSourceView(): void;
 };
 
+function parseEntryFromFluentSource(base: MessageEntry, source: string) {
+  const entry = parseEntry(source);
+  if (entry) {
+    entry.id = base.id;
+  }
+  return entry;
+}
+
+/**
+ * Create a new MessageEntry with a simple string pattern `value`,
+ * using `id` as its identifier.
+ */
+const createSimpleMessageEntry = (id: string, value: string): MessageEntry => ({
+  id,
+  value: {
+    type: 'message',
+    declarations: [],
+    pattern: { body: [{ type: 'text', value }] },
+  },
+});
+
 const initEditorData: EditorData = {
+  pk: 0,
   busy: false,
   entry: { id: '', value: null, attributes: new Map() },
-  fields: [],
-  focusField: { current: 0 },
-  initial: [],
+  focusField: { current: null },
+  initial: { id: '', value: null, attributes: new Map() },
   machinery: null,
+  fields: [],
   sourceView: false,
-  value: [],
 };
 
 const initEditorActions: EditorActions = {
   clearEditor: () => {},
   setEditorBusy: () => {},
   setEditorFromHelpers: () => {},
   setEditorFromHistory: () => {},
-  setEditorFromInput: () => {},
   setEditorSelection: () => {},
+  setResultFromInput: () => {},
   toggleSourceView: () => {},
 };
 
 export const EditorData = createContext(initEditorData);
+export const EditorResult = createContext<EditorResult>([]);
 export const EditorActions = createContext(initEditorActions);
 
+const buildResult = (message: EditorField[]): EditorResult =>
+  message.map(({ handle, keys, name }) => ({
+    name,
+    keys,
+    value: handle.current.value,
+  }));
+
 export function EditorProvider({ children }: { children: React.ReactElement }) {
   const locale = useContext(Locale);
   const { entity } = useContext(EntityView);
   const activeTranslation = useActiveTranslation();
   const readonly = useReadonlyEditor();
   const machinery = useContext(MachineryTranslations);
   const { setUnsavedChanges } = useContext(UnsavedActions);
-  const { exist } = useContext(UnsavedChanges);
-  const { resetFailedChecks } = useContext(FailedChecksData);
+  const { resetFailedChecks, source: failedChecksSource } =
+    useContext(FailedChecksData);
 
-  const [state, setState] = useState<EditorData>(initEditorData);
+  const [state, setState] = useState(initEditorData);
+  const [result, setResult] = useState<EditorResult>([]);
 
   const actions = useMemo<EditorActions>(() => {
```

Comment:
The actions that directly set a new value for `fields` need to also update `result`. Those that call `field.handle.current.setValue()` don't, as the EditField and EditAccessKey will take care of that.